### PR TITLE
PMM-5761 indexstats collector

### PR DESCRIPTION
**Pull request by percona-csalguero**
This branch is based on metrics backward compatibility because I needed the `walkTo` func implemented in that branch.

### CODE REVIEW

File: exporter/indexstats_collector.go [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+	client      *mongo.Client
+	collections []string
+}
+
+func (d *indexstatsCollector) Describe(ch chan<- *prometheus.Desc) {
+	prometheus.DescribeByCollect(d, ch)
+}
+
+func (d *indexstatsCollector) Collect(ch chan<- prometheus.Metric) {
+	ctx := context.TODO()
```
Comment (askomorokhov) [inline exporter/indexstats_collector.go:44]: It is not clear for me why and how long we should use `TODO()` context here.
Could you please add a comment.

**Review by askomorokhov — APPROVED** [review]
Looks good to me with comment. 👍

File: exporter/indexstats_collector.go (hunk 1 above)
Comment (BupycHuk) [reply exporter/indexstats_collector.go:44]: +1

File: exporter/indexstats_collector.go [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+			continue
+		}
+
+		var stats []bson.M
+		if err = cursor.All(ctx, &stats); err != nil {
+			logrus.Errorf("cannot get $indexStats for collection %s.%s: %s", database, collection, err)
+			continue
+		}
+
+		buf, _ := json.MarshalIndent(stats, "", "  ")
```
Comment (BupycHuk) [inline exporter/indexstats_collector.go]: looks different from what you did in #18

File: exporter/indexstats_collector.go [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+// create them with value 0 and the use walkTo to try to get the value.
+func sanitizeMetrics(m bson.M) bson.M {
+	ops := float64(0)
+	building := float64(0)
+
+	if val, ok := walkTo(m, []string{"accesses", "ops"}).(float64); ok {
+		ops = val
+	}
+	if val, ok := walkTo(m, []string{"building"}).(float64); ok {
+		ops = val
```
Comment (BupycHuk) [inline exporter/indexstats_collector.go]:
```suggestion
		building = val
```

File: exporter/indexstats_collector_test.go [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+		assert.NoError(t, err)
+	}
+
+	c := &indexstatsCollector{
+		client:      client,
+		collections: []string{"testdb.testcol_00", "testdb.testcol_01", "testdb.testcol_02"},
+	}
+
+	// The last \n at the end of this string is important
+	expected := strings.NewReader(`
```
Comment (BupycHuk) [inline exporter/indexstats_collector_test.go:68]: Why all values here are zero? Can we add something to change these values?

Comment (BupycHuk) [reply exporter/indexstats_collector_test.go:68]: If we can't let's add one more test for `sanitizeMetrics`

**Review by BupycHuk — CHANGES_REQUESTED** [review]

File: exporter/indexstats_collector_test.go [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+				},
+				"name": "idx_01",
+				"ns":   "testdb.testcol_01",
+				"v":    2,
+			},
+			"building": 1,
+		}
+		want := primitive.M{
+			"accesses": primitive.M{
+				"ops": float64(0),
```
Comment (BupycHuk) [inline exporter/indexstats_collector_test.go]: why values are 0 if it's 3 in input?

**Review by BupycHuk — CHANGES_REQUESTED** [review]
Looks like something wrong in `TestSanitize`

File: exporter/indexstats_collector_test.go (hunk 5 above)
Comment (percona-csalguero) [reply exporter/indexstats_collector_test.go]: Fixed

**Review by BupycHuk — CHANGES_REQUESTED** [review]
Please fix merge conflicts

**Review by BupycHuk — APPROVED** [review]

**Pull request merged**